client/index.jsx

```diff
@@ -18,174 +18,147 @@ function getDistance(lat1, lon1, lat2, lon2) {
   return d;
 }
 
-class Grass extends React.Component {
-  generateLink() {
-    let grass = this.props.grass;
-    let name = grass.name;
-    let plus = grass.plus_code ? "+" + grass.plus_code.compound_code : "";
+const Grass = ({coords, grass}) => {
+  const { name, plus_code, geometry: { location } } = grass;
+
+  const generateLink = () =>  {
+    const plus = plus_code ? "+" + plus_code.compound_code : "";
     return `https://www.google.com/maps/search/${name}${plus}`;
   }
-  computeDistance() {
-    let srcLoc = this.props.coords;
-    let destLoc = this.props.grass.geometry.location;
+
+  const computeDistance = () => {
+    const srcLoc = coords;
+    const destLoc = location;
     if (!srcLoc || !destLoc) return 0;
-    let lat1 = srcLoc.latitude;
-    let lon1 = srcLoc.longitude;
-    let lat2 = destLoc.lat;
-    let lon2 = destLoc.lng;
+    const {longitude: lon1, latitude: lat1} = srcLoc;
+    const {longitude: lat2, latitude: lon2} = srcLoc;
```

Comment:
This is supposed to be the destination.
```suggestion
    const {longitude: lon2, latitude: lat2} = destLoc;
```